app/src/main/java/com/github/gotify/MarkwonFactory.java

```diff
@@ -0,0 +1,99 @@
+package com.github.gotify;
+
+import android.content.Context;
+import android.graphics.Color;
+import android.graphics.Typeface;
+import android.text.style.BackgroundColorSpan;
+import android.text.style.BulletSpan;
+import android.text.style.QuoteSpan;
+import android.text.style.RelativeSizeSpan;
+import android.text.style.StyleSpan;
+import android.text.style.TypefaceSpan;
+import androidx.annotation.NonNull;
+import com.squareup.picasso.Picasso;
+import io.noties.markwon.AbstractMarkwonPlugin;
+import io.noties.markwon.Markwon;
+import io.noties.markwon.MarkwonSpansFactory;
+import io.noties.markwon.core.CorePlugin;
+import io.noties.markwon.core.CoreProps;
+import io.noties.markwon.ext.strikethrough.StrikethroughPlugin;
+import io.noties.markwon.ext.tables.TableAwareMovementMethod;
+import io.noties.markwon.ext.tables.TablePlugin;
+import io.noties.markwon.image.picasso.PicassoImagesPlugin;
+import io.noties.markwon.movement.MovementMethodPlugin;
+import org.commonmark.node.BlockQuote;
+import org.commonmark.node.Code;
+import org.commonmark.node.Emphasis;
+import org.commonmark.node.Heading;
+import org.commonmark.node.Link;
+import org.commonmark.node.ListItem;
+import org.commonmark.node.StrongEmphasis;
+
+public class MarkwonFactory {
+    public static Markwon create(Context context, Picasso picasso) {
+        return createBuilderBase(context, picasso).build();
+    }
+
+    public static Markwon createForNotification(Context context, Picasso picasso) {
+        final float[] headingSizes = {
+            2.F, 1.5F, 1.17F, 1.F, .83F, .67F,
+        };
+
+        final int bulletGapWidth =
+                (int) (8 * context.getResources().getDisplayMetrics().density + 0.5F);
+
+        return createBuilderBase(context, picasso)
+                .usePlugin(
+                        new AbstractMarkwonPlugin() {
+                            @Override
+                            public void configureSpansFactory(
+                                    @NonNull MarkwonSpansFactory.Builder builder) {
+                                builder.setFactory(
```

Comment:
To be honest, no. But I agree with you, if tables can't be displayed, at least the pure text should be visible. I've tried different things but nothing seems to help. I opened an issue in the Markwon project to get more info.

Reply:
I've slightly modified the [suggestion](https://github.com/noties/Markwon/issues/366). It would look like this:
![Screenshot_1629148710](https://user-images.githubusercontent.com/87871070/129631029-011e0083-fb9c-487b-b2d9-f3877b1be869.png)
I don't see any possibility of an even start of the cell text because the cells are only divided by a space character. So the table cells would drift away especially when the cell contents differ much in their size. Pure text is also possible. I'm not sure what the "right" or "more beautiful" solution is.

Reply:
I'd go with using pure text for the table representation.